src/test/java/com/baidu/hugegraph/api/BatchUpdateElementApiTest.java

```diff
@@ -0,0 +1,448 @@
+    }
+
+    @Test
+    public void testVertexBatchUpdateStrategyUnion() {
+        BatchVertexRequest req = batchVertexRequest("set", "old", "new",
+                                                    UpdateStrategy.UNION);
+
+        List<Vertex> vertices = vertexAPI.update(req);
+        assertBatchResponse(vertices, "set", "new", "old");
+
+        req = batchVertexRequest("set", "old", "old", UpdateStrategy.UNION);
+        vertices = vertexAPI.update(req);
+        assertBatchResponse(vertices, "set", "old");
+    }
+
+    @Test
+    public void testVertexBatchUpdateStrategyIntersection() {
+        BatchVertexRequest req = batchVertexRequest("set", "old", "new",
+                                                    INTERSECTION);
+
+        List<Vertex> vertices = vertexAPI.update(req);
+        assertBatchResponse(vertices, "set");
+
+        req = batchVertexRequest("set", "old", "old", INTERSECTION);
+        vertices = vertexAPI.update(req);
+        assertBatchResponse(vertices, "set", "old");
+    }
+
+    @Test
+    public void testVertexBatchUpdateStrategyAppend() {
+        BatchVertexRequest req = batchVertexRequest("list", "old", "old",
+                                                    UpdateStrategy.APPEND);
+
+        List<Vertex> vertices = vertexAPI.update(req);
+        assertBatchResponse(vertices, "list", "old", "old");
+
+        req = batchVertexRequest("list", "old", "new", UpdateStrategy.APPEND);
+        vertices = vertexAPI.update(req);
+        assertBatchResponse(vertices, "list", "old", "new");
+    }
+
+    @Test
+    public void testVertexBatchUpdateStrategyEliminate() {
+        BatchVertexRequest req = batchVertexRequest("list", "old", "old",
+                                                    UpdateStrategy.ELIMINATE);
+
+        List<Vertex> vertices = vertexAPI.update(req);
+        assertBatchResponse(vertices, "list");
+
+        req = batchVertexRequest("list", "old", "x", UpdateStrategy.ELIMINATE);
+        vertices = vertexAPI.update(req);
+        assertBatchResponse(vertices, "list", "old");
+    }
+
+    @Test
+    public void testVertexBatchUpdateWithInvalidArgs() {
+        BatchVertexRequest req = batchVertexRequest("set", "old", "old",
+                                                    UpdateStrategy.UNION);
+
+        Assert.assertThrows(ServerException.class, () -> {
+            List<Vertex> vertices = Whitebox.getInternalState(req, "vertices");
+            vertices.set(1, null);
+            Whitebox.setInternalState(req, "vertices", vertices);
+            vertexAPI.update(req);
+        });
+
+        Assert.assertThrows(ServerException.class, () -> {
+            Whitebox.setInternalState(req, "vertices", null);
+            vertexAPI.update(req);
+        });
+
+        Assert.assertThrows(ServerException.class, () -> {
+            Whitebox.setInternalState(req, "vertices", ImmutableList.of());
+            vertexAPI.update(req);
+        });
```

Comment:
add cases:
- vertices.size() > 500
- property type error like SUM strategy with string property, or UNION strategy with string property. (add a new test method for this cases)
- other exception cases